File: python/statementData.py
class statementData:
    def __init__(self, customer, performances):
        self.customer = customer
        # performances is the list of performanceGenerator objects
        self.performances: list = performances
        self.total_amount = self.get_total_amount()
        self.total_volume_credits = self.get_total_volume_credits()

    def get_total_amount(self):
        total_amount = 0
        for perf in self.performances:
            total_amount += perf.amount

        return total_amount

    def get_total_volume_credits(self):
        total_volume_credits = 0
        for performance in self.performances:
            total_volume_credits += performance.volume_credits

        return total_volume_credits

File: java/src/main/java/theatricalplays/StatementData.java
package theatricalplays;

import java.util.List;
import java.util.Map;

public final class StatementData {
    private final Invoice invoice;
    private final Map<String, Play> plays;

    public StatementData(Invoice invoice, Map<String, Play> plays) {
        this.invoice = invoice;
        this.plays = plays;
    }

    int totalAmount() {
        var totalAmount = 0;
        for (var perf : getPerformances()) {
            totalAmount += getPerformanceData(perf).amount();
        }
        return totalAmount;
    }

    int totalVolumeCredits() {
        var volumeCredits = 0;
        for (var perf : getPerformances()) {
            volumeCredits += getPerformanceData(perf).volumeCredits();
        }
        return volumeCredits;
    }

    Play playForPerformance(Performance perf) {
        return plays.get(perf.playID);
    }

    PerformanceData getPerformanceData(Performance perf) {
        return new PerformanceData(perf, plays.get(perf.playID));
    }

    List<Performance> getPerformances() {
        return invoice.performances;
    }

    public Map<String, Play> getPlays() {
        return plays;
    }

    public Invoice getInvoice() {
        return invoice;
    }
}